SnoopCompileCore/src/snoopi.jl

```diff
@@ -86,6 +94,57 @@ function _snoopi(cmd::Expr, tmin = 0.0)
     end
 end
 
+function start_deep_timing()
+    Core.Compiler.Timings.reset_timings()
+    Core.Compiler.__set_measure_typeinf(true)
+end
+function stop_deep_timing()
+    Core.Compiler.__set_measure_typeinf(false)
+    Core.Compiler.Timings.close_current_timer()
+end
+
+function finish_snoopi_deep()
+    return Core.Compiler.Timings._timings[1]
+end
+
+function _snoopi_deep(cmd::Expr)
+    return quote
+        start_deep_timing()
+        try
+            $(esc(cmd))
+        finally
+            stop_deep_timing()
+        end
+        finish_snoopi_deep()
+    end
+end
+
+macro snoopi_deep(cmd)
+    return _snoopi_deep(cmd)
+end
+
+function flatten_times(timing::Core.Compiler.Timings.Timing, tmin_secs = 0.0)
```

Comment:
Does this need to be here or could it be moved to SnoopCompile? I regard SnoopCompileCore as "the minimum you need to collect the data" and anything not strictly required should live in SnoopCompile. (The more logic we have here, the more it becomes a source of potential artifacts wrt invalidation etc.)

Reply:
Agreed. I think we should move it! :)

I had it here because in an earlier version i was calling this before returning; similar to `sort_timed_inf()`.

Thanks. Leaving this as a todo

Reply:
Yes, definitely should be moved!

Reply:
Thanks Done.